### PR TITLE
Move the compiler toolchain in it's own webworker.

### PR DESCRIPTION
**Pull request by Kuinox**
This is the first iteration towards having the user program run in a separated process.  
While this doesn't achieve it, the entier toolchain is now in a webworker, which mean a separated process.  
Typing should no longer freeze the whole interface.
The worker JS code is fully reusable without any change so there shouldn't be more difficulties due to the JS ecosystem...

### CODE REVIEW

File: src/Draco.Editor.Web/OnInit.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace Draco.Editor.Web;
+
+public class OnInit
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/OnInit.cs:3]: Is sealed applicable here?

File: src/Draco.Editor.Web/app/build.js [hunk 2]
```diff
@@ -7,11 +7,50 @@ import { createThemeBasedLogo as getFavicon } from './favicon-downloader.js';
...
+    // Problem #2: I lost 30 mins trying to configure rollup to output `dotnet.js` instead of `dotnet.cjs` without success.
+    // So I rename the file by hand instead...
+    fs.renameSync(path.join(outDir, 'dotnet.umd.cjs'), path.join(outDir, 'dotnet.js'));
+
+    fs.copyFileSync(path.join(binFolder, 'dotnet.wasm'), path.join(outDir, 'dotnet.wasm'));
+}
+
+await dotnetjsBuild();
+
+
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/app/build.js:55]: Lot of blank lines, maybe do an autoformat on this file?

File: src/Draco.Editor.Web/app/ts/app.ts [hunk 3]
```diff
@@ -9,11 +9,7 @@ import { deflateRaw, inflateRaw } from 'pako';
 // This file is run on page load.
 // This run before blazor load, and will tell blazor to start.
 
-declare global { // Blazor do not provide types, so we have our own to please typescript.
-    class Blazor {
-        static start(): Promise<void>;
-    }
-}
+const worker = new Worker('worker.js'); // first thing: we start the worker so it load in parallel.
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/app/ts/app.ts:12]:
```suggestion
const worker = new Worker('worker.js'); // first thing: we start the worker so it loads in parallel.
```

File: src/Draco.Editor.Web/app/ts/app.ts [hunk 4]
```diff
@@ -132,16 +123,48 @@ const outputEditor = monaco.editor.create(document.getElementById('output-viewer
...
+    switch (msg.type) {
+    case 'setOutputText':
+        outputEditor.getModel().setValue(msg.message);
+        break;
+    default:
+        break;
+    }
+};
+
+
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/app/ts/app.ts:140]: Double bank line, maybe do autoformat?

File: src/Draco.Editor.Web/app/ts/worker.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
+importScripts('./dotnet.js');
+declare global { // Blazor do not provide types, so we have our own to please typescript.
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/app/ts/worker.ts:2]:
```suggestion
declare global { // Blazor does not provide types, so we have our own to please typescript.
```

File: src/Draco.Editor.Web/app/ts/app.ts (hunk 4 above)
Comment (Kuinox) [reply src/Draco.Editor.Web/app/ts/app.ts:140]: autoformat does not remove blanks lines.

**Review by LPeter1997 — APPROVED** [review]
LGTM

**Pull request merged**